### PR TITLE
BUGFIX: Ensure .md links with hashes are rewritten properly

### PR DESCRIPTION
**Pull request by unclecheese**
Resolves: https://github.com/silverstripe/silverstripe-userhelp-content/issues/145

### CODE REVIEW

File: src/utils/rewriteLink.ts [hunk 1]
```diff
@@ -102,6 +101,22 @@ const rewriteLink = (
                 domToReact(children, parseOptions)
             )
         }
+        const hashMatches = href.match(/(\.md#([A-Za-z0-9_-]+))$/);
```
Comment (emteknetnz) [inline src/utils/rewriteLink.ts:104]: I think you'll need to escape the final `-`
```suggestion
        const hashMatches = href.match(/(\.md#([A-Za-z0-9_\-]+))$/);
```

**Comment by emteknetnz** [comment]
How come this PR doesn't auto-deploy on the CI like this PR does https://github.com/silverstripe/doc.silverstripe.org/pull/226?

File: src/utils/rewriteLink.ts (hunk 1 above)
Comment (unclecheese) [reply src/utils/rewriteLink.ts:104]: I don't believe you do? I mean, it's working in the preview: https://5efaa115098fad000928d5d0--ss-userhelp.netlify.app/en/4/optional_features/content_blocks/

You can see the raw source here: https://github.com/dnadesign/silverstripe-elemental/blame/4/docs/en/userguide/index.md#L6 where the `.md` is in the link and removed on the rendered docs site.

Comment (emteknetnz) [reply src/utils/rewriteLink.ts:104]: ok all good then

**Pull request merged**